odoo/addons/base/views/ir_qweb_widget_templates.xml

```diff
@@ -28,12 +28,22 @@
                 <i t-if="not options.get('no_marker')" class="fa fa-map-marker fa-fw" role="img" aria-label="Address" title="Address"/>
                 <span class="w-100 lh-sm text-truncate d-block" itemprop="streetAddress" t-esc="address"/>
             </div>
-            <div t-if="city and 'city' in fields" t-attf-class="d-flex align-items-baseline gap-1">
+            <t t-set="has_city" t-value="city and 'city' in fields"/>
+            <t t-set="has_state" t-value="object.state_id and object.country_id.state_required"/>
+            <div t-if="has_city or (country_id and 'country_id' in fields)" t-attf-class="d-flex align-items-baseline gap-1">
                 <i t-if="not options.get('no_marker')" class="fa fa-map-marker fa-fw" role="img" aria-label="Address" title="Address"/>
                 <span>
                     <div>
-                        <span itemprop="addressLocality" t-esc="city"/>,
-                        <span itemprop="addressCountry" t-esc="country_id"/>
+                        <t t-if="has_city and country_id">
+                            <span itemprop="addressLocality" t-out="city"/>,
+                        </t>
+                        <t t-elif="has_city">
+                            <span itemprop="addressLocality" t-out="city"/>
+                        </t>
```

Comment:
Maybe
```suggestion
                        <t t-if="has_city">
                            <span itemprop="addressLocality" t-out="city"/><t t-if="has_state or country_id">,</t>
                        </t>
```

Reply:
Hello @tde-banana-odoo done with the changes 
`t-if="has_state or country_id"` I think here we can use country_id directly because has_state will be true only when country_id is present.

Reply:
Quite true :D